upup/pkg/fi/cloudup/awstasks/external_target_group_attachment.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awstasks
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/golang/glog"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/cloudformation"
+	"k8s.io/kops/upup/pkg/fi/cloudup/terraform"
+)
+
+//go:generate fitask -type=ExternalTargetGroupAttachment
+type ExternalTargetGroupAttachment struct {
+	Name      *string
+	Lifecycle *fi.Lifecycle
+
+	TargetGroupARN string
+
+	AutoscalingGroup *AutoscalingGroup
+}
+
+func (e *ExternalTargetGroupAttachment) name() string {
+	re := regexp.MustCompile("arn:.*:targetgroup/")
+	return re.ReplaceAllString(*e.Name, "")
```

Comment:
It's probably better to set the name field with this value when we create the ExternalTargetGroupAttachment.  But that can be moved later if we need it, I think.